crates/orchestrator/src/tests/data_storage/mod.rs

```diff
@@ -0,0 +1,43 @@
+use crate::data_storage::aws_s3::config::AWSS3Config;
+use crate::data_storage::aws_s3::AWSS3;
+use crate::data_storage::{DataStorage, DataStorageConfig};
+use crate::tests::config::TestConfigBuilder;
+use bytes::Bytes;
+use rstest::rstest;
+use serde_json::json;
+use utils::env_utils::get_env_var_or_panic;
+
+/// This test checks the ability to put and get data from AWS S3 using `AWSS3`.
+/// It puts JSON data into a test bucket and retrieves it, verifying the data
+/// matches what was originally uploaded.
+/// Dependencies: `color_eyre`, `dotenvy`, `rstest`, `tokio`, `serde_json`.
+#[rstest]
+#[tokio::test]
+async fn test_put_and_get_data_s3() -> color_eyre::Result<()> {
+    TestConfigBuilder::new().build().await;
```

Comment:
`TestConfigBuilder` allows for customisation over any `external client` of our choice,
moving it to a global fixture is not feasible since all test cases have different customisation requirements.

We can make fixture for tests under same scope if they require same customised external clients.

for eg : 
All tests under `da_job` if require same config customisation, can implement a `fixture` just for themselves.
similarly for other scopes .

We can create a separate issue for this and resolve there. 


### IMPORTANT 
[This branch is a fork from [increasing coverage](https://github.com/madara-alliance/madara-orchestrator/tree/feat/increasing-test-coverage) ]
[This comment refers to changed already merged to main, they are shown here since #52 is merged and the base of this PR automatically shifted to `main` ]